### ISSUE TITLE
false offense detected by Style/MixinUsage

### ISSUE THREAD

**Issue by InteNs** [bbatsov/rubocop#4885]
## Expected behavior

```ruby
thing = Class.new do
  attr_accessor :value_text, :value_type
  include IsNameValueType
end

RSpec.describe IsNameValueType do
  subject { thing.new }
  # tests
end
```

expect Mixin usage to be correct

## Actual behavior

```
Inspecting 4 files
.C..

Offenses:

spec/models/concerns/is_name_value_type_spec.rb:3:3: C: Style/MixinUsage: include is used at the top level. Use inside class or module.
  include IsNameValueType
  ^^^^^^^^^^^^^^^^^^^^^^^

4 files inspected, 1 offense detected
```

## Steps to reproduce the problem

create the test as in the expected behaviour and run `rubocop <filename>`

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
➔ rubocop -V
0.51.0 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-darwin16)
```

**Comment by dreid** [comment]
It looks like this cop misreports for any case where the class is the first in some set of siblings.

For example:

```ruby
class A
  include Mixin
end

class B
  include Mixin
end
```

Result:

```
❯ rubocop t.rb
Inspecting 1 file
C

Offenses:

t.rb:2:3: C: include is used at the top level. Use inside class or module.
  include Mixin
  ^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

Of note is that only the first class has this error reported.

```
❯ rubocop -V
0.51.0 (using Parser 2.4.0.0, running on ruby 2.3.4 x86_64-darwin16)
```

**Comment by benhutton** [comment]
I'm also getting false positives for this:

```
RSpec::Matchers.define :do_something do
  include Mixin
end
```

Again, strangely, only for the first occurrence in a file. I define three matchers in one file, all with the same mixin, and only get flagged for the first one.

.....

It is a bit unclear from the [mixin code itself](https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/cop/style/mixin_usage.rb) and the [PR](https://github.com/bbatsov/rubocop/pull/4840) what the goal of the cop is. Is it really to prevent mixing into `Object`? Is it to prevent mixing into `main`? Is it to prevent mixing in outside of a class or module? Let's state very explicitly what the goal is, and then implement against it more narrowly!

**Comment by RemyMaucourt** [comment]
I have the same problem, and another info.

    class A
      include module_A
      include namespace_1::module_B
      include namespace_1::module_C
      include module_D
    end

    class B
       include module_E
    end

Rubocop will raise Style/MixinUsage only for module_A and module_D, not the namespaced ones.

**Comment by Drenmi** [comment]
Indeed. It's a bit unclear what the implementation is trying to do, but it's clear that it's incorrect. 🙂

**Comment by ZASMan** [comment]
Is there any fix on this? It's still popping up for me.

**Comment by pocke** [comment]
@ZASMan 
Can you try updating rubocop to 0.52.0? It has been fixed in 0.52.0 already.
If you still have any bugs, please open a new issue.
